frontend/packages/network-attachment-definition-plugin/integration-tests/tests/models/nadForm.ts

```diff
@@ -0,0 +1,68 @@
+import { browser, ElementFinder, ExpectedConditions as until } from 'protractor';
+import { BROWSER_TIMEOUT, testName } from '@console/internal-integration-tests/protractor.conf';
+import { createYAMLButton, isLoaded } from '@console/internal-integration-tests/views/crud.view';
+import { PAGE_LOAD_TIMEOUT_SECS } from '@console/kubevirt-plugin/integration-tests/tests/utils/consts';
+import { click, selectDropdownOptionById } from '@console/shared/src/test-utils/utils';
+import * as nadFormView from '../../views/nad.form.view';
+
+// TODO See about moving this function from kubevirt-plugin to console-shared
+async function fillInput(elem: ElementFinder, value: string) {
+  // Sometimes there seems to be an issue with clear() method not clearing the input
+  let attempts = 3;
+
+  /* eslint-disable no-await-in-loop */
+  do {
+    --attempts;
+    if (attempts < 0) {
+      throw Error(`Failed to fill input with value: '${value}'.`);
+    }
+    await browser.wait(until.and(until.presenceOf(elem), until.elementToBeClickable(elem)));
+    // TODO: line below can be removed when pf4 tables in use.
+    await elem.click();
+    await elem.clear();
+    await elem.sendKeys(value);
+  } while ((await elem.getAttribute('value')) !== value && attempts > 0);
+  /* eslint-enable no-await-in-loop */
+}
+
+export class NADForm {
+  async openNADForm() {
+    await click(createYAMLButton);
+    await browser.sleep(500); // wait until the fade in effect is finished TODO check if needed
```

Comment:
this wait should not be necessary for the in-page forms

Reply:
Done!